scripts/check_python.py

```diff
@@ -0,0 +1,132 @@
+#!/usr/bin/env python3
+
+import argparse
+import multiprocessing
+import os
+import subprocess
+import sys
+from concurrent import futures
+from pathlib import Path
+from typing import List
+
+
+def run_cmd(cmd, verbose):
+    if verbose:
+        print(f"Running: {cmd}")
+    completed = subprocess.run(cmd, stdout=subprocess.PIPE, stderr=subprocess.PIPE, check=False, encoding="UTF-8")
+    return completed
+
+
+if __name__ == "__main__":
+    # Parse command-line arguments
+    parser = argparse.ArgumentParser(description="Check or fix Python like code.")
+    parser.add_argument(
+        "--fix",
+        "-fix",
+        "-f",
+        default=False,
+        action="store_true",
+        help="Fix the sources instead of just checking their formatting.",
+    )
+    parser.add_argument(
+        "--verbose", "-v", default=False, action="store_true", help="Verbosely print what I am doing.",
+    )
+    parser.add_argument(
+        "--pylint", default=False, action="store_true", help="Run pylint and other explicitly requested commands.",
+    )
+    parser.add_argument(
+        "--black", default=False, action="store_true", help="Run black and other explicitly requested commands.",
+    )
+    parser.add_argument(
+        "--isort", default=False, action="store_true", help="Run isort and other explicitly requested commands.",
+    )
+    parser.add_argument(
+        "--exclude",
+        default=[],
+        action="append",
+        help="Add a pattern to exclude from processing. The pattern can be a path or a glob.",
+    )
+    parser.add_argument("roots", type=str, nargs="+", help="The root paths to search from, or files to check.")
+    args = parser.parse_args()
+
+    if not args.pylint and not args.black and not args.isort:
+        args.pylint = True
+        args.black = True
+        args.isort = True
+
```

Comment:
Somewhere after these lines, can we print to console the `pylint`, `black` and `isort` versions used? This will help formatting the code locally with exactly the same versions, otherwise different versions may format the code differently and this would cause CI failures.

Reply:
:-) OK. I don't like that it prints those versions. I'll add a flag to *disable* printing the versions, but print them by default.